### PR TITLE
Use Java 23 as latest with more consistency

### PR DESCRIPTION
**Pull request by akurtakov**

### CODE REVIEW

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java [hunk 1]
```diff
@@ -404,7 +404,7 @@ private AST createAST(Map<String, String> options, int level, Context context) {
 		long sourceLevel = CompilerOptions.versionToJdkLevel(sourceModeSetting);
 		if (sourceLevel == 0) {
 			// unknown sourceModeSetting
-			sourceLevel = ClassFileConstants.JDK21; // TODO latest
+			sourceLevel = ClassFileConstants.JDK23; // TODO latest
```
Comment (mickaelistria) [inline org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java]: Isn't there some "latest" we can reference here?

File: org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/AST.java [hunk 2]
```diff
@@ -534,7 +534,7 @@ public final class AST {
 	 * Internal property for latest supported JLS level
 	 * This provides the latest JLS level.
 	 */
-	private static final int JLS_INTERNAL_Latest = JLS22;
+	private static final int JLS_INTERNAL_Latest = JLS23;
```
Comment (mickaelistria) [inline org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/AST.java]: Should we wait for that to change in JDT?

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java (hunk 1 above)
Comment (akurtakov) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java]: Fixed now.

File: org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/AST.java (hunk 2 above)
Comment (akurtakov) [reply org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/AST.java]: I do this change as without it there are test failures like https://ci.eclipse.org/ls/job/jdt-core-incubator/job/dom-with-javac/140/testReport/junit/org.eclipse.jdt.core.tests.dom/ASTTest/Build_and_Test___testASTLevels___JLS2/

Comment (akurtakov) [reply org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/AST.java]: This fixed the mentioned tests but caused others to fail thus removed from this PR.

**Pull request merged**